src/framework/actions/base.ts

```diff
@@ -0,0 +1,24 @@
+import { CardID, GameState } from "../types";
+
+export const buyWorker = ($: GameState, cid: CardID): void => {
+  const P = $.players[$.activePlayer];
+
+  if (P.hasBuiltWorkerThisTurn) {
+    throw new Error("only one worker per turn");
+  }
+
+  if (P.gold < 1) {
+    throw new Error("cannot afford worker");
+  }
+
+  const indexInHand = P.hand.indexOf(cid);
+  if (indexInHand < 0) {
+    throw new Error("card not in hand");
+  }
+
+  P.hand = P.hand.slice(0, indexInHand).concat(P.hand.slice(indexInHand + 1));
```

Comment:
This happens often enough we should really make a helper for it; something like `removeCard`, which maybe returns false or throws when the card doesn't exist? Actually, don't we have that somewhere?

Reply:
Yeah, we do it in at least one other place, but it comes with some extra side effects. Having it be a dedicated function is a good idea.